client/src/components/friendModal/ViewFriendItem.jsx

```diff
@@ -12,22 +19,72 @@ const useStyles = makeStyles((theme) => ({
     color: theme.palette.secondary.light,
     backgroundColor: theme.palette.primary.main,
   },
+  acceptedFriendButton: {
+    borderRadius: 25,
+    color: theme.palette.secondary.dark,
+    backgroundColor: theme.palette.primary.main,
+  },
 }));
 
-const ViewFriendItem = ({ friend, typeOfFriendRequest }) => {
+const ViewFriendItem = ({ friend, typeOfFriendRequest, refresh, setRefresh }) => {
   const classes = useStyles();
 
   const [clicked, setClicked] = useState(false);
+  const [rejectClicked, setRejectClicked] = useState(false);
 
-  /** Change the primary/clicked colour of the button depending on which tab we're in*/
+  /** Change the primary/clicked colour of the button depending on which tab we're in */
   function resolveTypeOfButton() {
     if (typeOfFriendRequest === 'Friends') return clicked ? classes.addFriendButton : classes.removeFriendButton;
+    else if (typeOfFriendRequest === 'Received')
+      return clicked ? classes.acceptedFriendButton : classes.addFriendButton;
     else return clicked ? classes.removeFriendButton : classes.addFriendButton;
   }
 
-  const handleAdd = (id) => {
-    //TODO: integrate backend to make friend request
+  const handleClick = async () => {
+    let response;
+    //The button is not disabled when clicked, so we need to handle both states of "clicked"
+    if (typeOfFriendRequest === 'Suggested') {
+      if (!clicked) {
+        response = await postNewFriendRequest(friend._id);
+        if (response.status === 200) {
+          console.log('Added new friend');
+        } else {
+          console.log(response.data);
+        }
+      } else {
+        response = await deleteOutgoingFriendRequest(friend._id);
+        if (response.status === 200) {
+          console.log('Deleted friend request');
+        } else {
+          console.log(response.data);
+        }
+      }
+    } else if (typeOfFriendRequest === 'Friends') {
+      response = await deleteFriend(friend._id);
+      if (response.status === 200) {
+        console.log('Deleted friend');
+      } else {
+        console.log(response.data);
+      }
+    } else {
+      response = await postAcceptFriendRequest(friend._id);
```

Comment:
What happens if you click these 2x?

Reply:
The button is disabled in the front end so it should be impossible to double click. For the outgoing Friend Request, once the request was deleted, the button changes back to making a friend request so you'll have to make another friend request before deleting. You can't click twice in a row